credentials/tls.go

```diff
@@ -243,14 +243,14 @@ var cipherSuiteLookup = map[uint16]string{
 	tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305:  "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
 }
 
-// cloneTLSConfig returns a shallow clone of the exported
+// CloneTLSConfig returns a shallow clone of the exported
 // fields of cfg, ignoring the unexported sync.Once, which
 // contains a mutex and must not be copied.
 //
 // If cfg is nil, a new zero tls.Config is returned.
 //
 // TODO: inline this function if possible.
-func cloneTLSConfig(cfg *tls.Config) *tls.Config {
+func CloneTLSConfig(cfg *tls.Config) *tls.Config {
```

Comment:
Why are you exporting this function?  It says TODO: inline if possible; I don't think it was intended to be part of our public API.

Reply:
Can we move this to credential/internal instead? aTLS also defined this function(https://github.com/grpc/grpc-go/blob/b830b5f361072689b94d99055e8ffc51821eabf3/security/advancedtls/advancedtls.go#L503-L508), and ideally it's good to share this code. But yeh I just noticed that the code length was short enough and we could also inline it. Can I know which way do you prefer?

Reply:
It should be fine to move it to `internal` and export it.  But we don't want this in our public API.

Reply:
How about function `AppendH2ToNextProtos` then? I also changed it to an exported function in `tls.go` because there is a duplicate in `advancedtls.go`. Should I move both functions to `/credential/interval/` to export them there?

Reply:
@dfawley Sure! I was not intended to make the API public. We will move it to internal and export it.
@cindyxue internal packages is kind of special that even if you export the function(capitalize the function's first character), it is not public API, but we can use that to shared the code between aTLS module and main gRPC module. You can find more [here](https://golang.org/doc/go1.4#internalpackages).